### PR TITLE
Use glow instead of sparkle

### PR DESCRIPTION
**Pull request by sagudev**
[sparkle](https://github.com/servo/sparkle) is servo owned (but mostly unmaintained) gl create that we would like to archive and replace with [glow](https://github.com/grovesNL/glow/tree/main), that is actively maintained and used by wgpu and wider rust community.

Part of https://github.com/servo/servo/issues/33539

### CODE REVIEW

File: webxr/Cargo.toml [hunk 1]
```diff
@@ -33,7 +33,8 @@ euclid = "0.22"
 log = "0.4.6"
 openxr = { version = "0.19", optional = true }
 serde = { version = "1.0", optional = true }
-sparkle = "0.1"
+#glow = "0.14"
+glow = { git = "https://github.com/sagudev/glow.git", branch = "get_parameter_bool_array" }
```
Comment (sagudev) [inline webxr/Cargo.toml:37]: we need https://github.com/grovesNL/glow/pull/308 released

File: webxr/openxr/mod.rs [hunk 2]
```diff
@@ -799,7 +801,7 @@ impl LayerManagerAPI<SurfmanGL> for OpenXrLayerManager {
                     })?;
                 let color_texture = device.surface_texture_object(color_surface_texture);
                 let color_target = device.surface_gl_texture_target();
-                let depth_stencil_texture = openxr_layer.depth_stencil_texture;
+                let depth_stencil_texture = openxr_layer.depth_stencil_texture.map(|nt| nt.0.get());
```
Comment (mrobinson) [inline webxr/openxr/mod.rs:804]:
```suggestion
                let depth_stencil_texture = openxr_layer.depth_stencil_texture.map(|texture| texture.0.get());
```

File: webxr/surfman_layer_manager.rs [hunk 3]
```diff
@@ -163,7 +174,7 @@ impl LayerManagerAPI<SurfmanGL> for SurfmanLayerManager {
                     .iter()
                     .map(|&viewport| SubImage {
                         color_texture,
-                        depth_stencil_texture,
+                        depth_stencil_texture: depth_stencil_texture.map(|nt| nt.0.get()),
```
Comment (mrobinson) [inline webxr/surfman_layer_manager.rs:177]:
```suggestion
                        depth_stencil_texture: depth_stencil_texture.map(|texture| texture.0.get()),
```

**Review by mrobinson — APPROVED** [review]

**Comment by sagudev** [comment]
per https://github.com/servo/servo/pull/33538#issuecomment-2418618418 we will use unreleased surfman version for now

can you merge this @mrobinson

**Pull request merged**